test/graph_generation_tests.cc

```diff
@@ -1459,5 +1459,57 @@ namespace detail {
 		maybe_print_graphs(ctx);
 	}
 
+	TEST_CASE("side effects generate appropriate command-dependencies", "[graph_generator][command-graph][side-effect]") {
+		using order = experimental::side_effect_order;
+		static constexpr auto side_effect_orders = {order::sequential};
+		constexpr size_t num_nodes = 2;
+
+		// TODO placeholder: complete with dependency types for other side effect orders
+		const auto expected_dependencies = std::unordered_map<std::pair<order, order>, std::optional<dependency_kind>, pair_hash>{
+		    {{order::sequential, order::sequential}, dependency_kind::TRUE_DEP},
+		};
+
+		const auto order_a = GENERATE(values(side_effect_orders));
+		const auto order_b = GENERATE(values(side_effect_orders));
+		CAPTURE(order_a);
+		CAPTURE(order_b);
+
+		test_utils::cdag_test_context ctx(num_nodes);
+		auto& tm = ctx.get_task_manager();
+		auto& ggen = ctx.get_graph_generator();
+		test_utils::mock_host_object_factory mhof;
+
+		auto ho_common = mhof.create_host_object(); // should generate dependencies
+		auto ho_a = mhof.create_host_object();      // should NOT generate dependencies
+		auto ho_b = mhof.create_host_object();      // -"-
```

Comment:
What is the purpose of `ho_a` and `ho_b` if a dependency is always created through `ho_common`?

Reply:
Good catch! I've added one of the side effects to another task to detect incorrect dependencies.